### PR TITLE
Provide an OpenSSF security insights manifest file

### LINKED ISSUE #27038: Inclusion of security-insights manifest for enhanced clomonitor.io
### Description

To align with CNCF best practices and increase our project's health metrics on [CLOMonitor](https://clomonitor.io/search?text=keycloak&page=1), we need to adopt the [Security Insights Specification](https://github.com/ossf/security-insights-spec/blob/v1.0.0/specification.md). 

This approach allows us to communicate security-related insights with our community. Given our status as an incubating CNCF project, integrating the SECURITY-INSIGHTS manifest will demonstrate our commitment to security and open-source health. This move not only aligns us with industry standards but also contributes to a safer open-source ecosystem.

### PR DESCRIPTION
**Pull request by abstractj**
Closes #27038

### CODE REVIEW

**Review by rmartinc — APPROVED** [review]
I have not even heard about this file. :smile:
@abstractj Very good idea!

**Review by jonkoops — DISMISSED** [review]

File: SECURITY-INSIGHTS.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+header:
+  schema-version: 1.0.0
+  expiration-date: '2025-02-16T01:00:00.000Z'
```
Comment (stianst) [inline SECURITY-INSIGHTS.yml:3]:
```suggestion
  expiration-date: '2025-02-14T01:00:00.000Z'
```

Should be max 1 year from last-reviewed

File: SECURITY-INSIGHTS.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  last-updated: '2024-02-14'
+  last-reviewed: '2024-02-14'
+  project-url: 'https://github.com/keycloak/keycloak'
+  license: 'https://github.com/keycloak/keycloak/blob/main/LICENSE.txt'
+project-lifecycle:
+  bug-fixes-only: false
+  core-maintainers:
+    - https://github.com/keycloak/keycloak/blob/main/MAINTAINERS.md
+  status: active
```
Comment (stianst) [inline SECURITY-INSIGHTS.yml:12]:
```suggestion
  status: Active
```

File: SECURITY-INSIGHTS.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      action: allowed
+      path:
+        - /
+  contributing-policy: 'https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md'
+  code-of-conduct:
+  - 'https://github.com/keycloak/keycloak?tab=coc-ov-file'
+documentation:
+  - 'https://www.keycloak.org/documentation'
+distribution-points:
+  - 'https://www.keycloak.org/downloads'
```
Comment (stianst) [inline SECURITY-INSIGHTS.yml:27]: Should we add Quay containers? NPM packages? GitHub Releases?

**Review by stianst — APPROVED** [review]

**Pull request merged**